test/unit/core/input.js

```diff
@@ -0,0 +1,26 @@
+import Input from '../../../modules/input';
+
+describe('subscriptions', function () {
+  it('subscribtion on beforeinput event should be', function () {
+    let addEventListenerCallCount = 0;
+    let eventName = '';
+    let handler;
+    const quill = {
+      on() {},
+      root: {
+        addEventListener(name, callback) {
+          eventName = name;
+          handler = callback;
+          addEventListenerCallCount += 1;
+        },
+      },
+    };
+
+    // eslint-disable-next-line no-new
+    new Input(quill, {});
+
+    expect(addEventListenerCallCount).toEqual(1);
+    expect(eventName).toEqual('beforeinput');
+    expect(typeof handler).toEqual('function');
+  });
+});
```

Comment:
I see some kind of an "integration" test in `quill.js` for `beforeinput` event but don't see a similar one for `compositionStart`. Cannot we add it?